### ISSUE TITLE
Crypttab broken after upgrading to Mint 20.1

### ISSUE THREAD

**Issue by rsi9x** [linuxmint/linuxmint#349]
I looks like the crypttab is broken in Mint 20.1.  I upgraded from Mint 20 to Mint 20.1, and was unable to boot with two LUKS volumes in crypttab.  If the second volume listed in crypttab doesn't have a password file specified (set to none), it will prompt for the password for the first volume, then hang.  If you create a password file and assign it to the second volume, then the system boots.

Hangs:
```
nvme0n1p6_crypt UUID=d62d9482-3ffe-4dfb-b72d-9861effcc596 none luks,discard
data UUID=c388a9bf-a5d5-495f-8c83-1738e9154b07 none luks,discard
```

Boots:
```
nvme0n1p6_crypt UUID=d62d9482-3ffe-4dfb-b72d-9861effcc596 none luks,discard
data UUID=c388a9bf-a5d5-495f-8c83-1738e9154b07 /root/data.key luks,discard
```

Both volumes have the same passphrase. Prior to the upgrade, after it prompted for the passphrase for the first volume, it would automatically unlock the second volume, since the passphrase was the same. I also tried changing the passphrase of the second volume, and it didn't prompt me for a different passphrase to unlock the second volume.

I've been able to reproduce this issue on both a desktop and a laptop with similar disk configurations. Others are reporting similar experiences in the forums. [https://forums.linuxmint.com/viewtopic.php?f=46&t=340309](url)

Other relevant info:

```
System:    Kernel: 5.4.0-62-generic x86_64 bits: 64 compiler: gcc v: 9.3.0 Desktop: Cinnamon 4.8.6 wm: muffin 4.8.1 
           dm: LightDM 1.30.0 Distro: Linux Mint 20.1 Ulyssa base: Ubuntu 20.04 focal 
CPU:       Topology: 8-Core model: AMD Ryzen 7 2700X bits: 64 type: MT MCP arch: Zen+ rev: 2 L2 cache: 4096 KiB 
           flags: avx avx2 lm nx pae sse sse2 sse3 sse4_1 sse4_2 sse4a ssse3 svm bogomips: 118386 
           Speed: 2094 MHz min/max: 2200/3700 MHz boost: enabled Core speeds (MHz): 1: 2094 2: 1960 3: 2384 4: 2566 5: 2035 
           6: 2236 7: 2297 8: 2187 9: 3883 10: 2398 11: 2092 12: 2184 13: 2398 14: 2434 15: 2074 16: 1993 
Graphics:  Device-1: Advanced Micro Devices [AMD/ATI] Navi 10 [Radeon RX 5600 OEM/5600 XT / 5700/5700 XT] 
           vendor: Micro-Star MSI driver: amdgpu v: kernel bus ID: 0b:00.0 chip ID: 1002:731f 
           Display: x11 server: X.Org 1.20.9 driver: amdgpu,ati unloaded: fbdev,modesetting,radeon,vesa alternate: nvidia 
           resolution: 1600x1200~60Hz, 1600x1200~60Hz, 1920x1200~60Hz 
           OpenGL: renderer: AMD Radeon RX 5700 XT (NAVI10 DRM 3.35.0 5.4.0-62-generic LLVM 11.0.1) 
           v: 4.6 Mesa 20.3.3 - kisak-mesa PPA direct render: Yes 
Drives:    Local Storage: total: 1.47 TiB used: 510.16 GiB (33.8%) 
           ID-1: /dev/nvme0n1 vendor: Western Digital model: WDS500G3X0C-00SJG0 size: 465.76 GiB speed: 31.6 Gb/s lanes: 4 
           serial: <filter> rev: 111110WD scheme: MBR 
           ID-2: /dev/sda vendor: Patriot model: Pyro size: 111.79 GiB speed: 6.0 Gb/s serial: <filter> rev: BBF0 scheme: GPT 
           ID-3: /dev/sdb vendor: Samsung model: SSD 860 EVO 1TB size: 931.51 GiB speed: 6.0 Gb/s serial: <filter> rev: 1B6Q 
Partition: ID-1: / size: 455.29 GiB used: 50.33 GiB (11.1%) fs: ext4 dev: /dev/dm-1 
           ID-2: /boot size: 703.5 MiB used: 305.8 MiB (43.5%) fs: ext4 dev: /dev/nvme0n1p5 
           ID-3: swap-1 size: 980.0 MiB used: 0 KiB (0.0%) fs: swap dev: /dev/dm-2 
           
           
    NAME                MAJ:MIN RM   SIZE RO TYPE  MOUNTPOINT
nvme0n1             259:0    0 465.8G  0 disk  
├─nvme0n1p1         259:1    0   512M  0 part  /boot/efi
├─nvme0n1p2         259:2    0     1K  0 part  
├─nvme0n1p5         259:3    0   731M  0 part  /boot
└─nvme0n1p6         259:4    0 464.6G  0 part  
  └─nvme0n1p6_crypt 253:0    0 464.5G  0 crypt 
    ├─vgmint-root   253:1    0 463.6G  0 lvm   /
    └─vgmint-swap_1 253:2    0   980M  0 lvm   [SWAP]
...
sdb                   8:16   0 931.5G  0 disk  
└─data              253:3    0 931.5G  0 crypt 
  ├─data-home       253:4    0   500G  0 lvm   /var/local/data/home
  └─data-common     253:5    0    10G  0 lvm   /var/local/data/common
```

**EDIT**: It appears that the issue is with plymouth and affects any non-OS volumes. Once the OS starts booting, plymouth seems to fail to identify when to prompt the user for a passphrase, causing the boot process to hang.  Disabling plymouth by removing 'quiet splash' from the kernel boot parameters seems to be a temporary workaround for this issue.

**Comment by feikede** [comment]
Sorry, forgot some files:

```
rainer@cslbox:~ $ cat /etc/fstab 
# /etc/fstab: static file system information.
UUID=32ba72e3-26fe-441d-9190-38165e662d4c /               ext4    errors=remount-ro 0       1
UUID=A923-55B5  /boot/efi       vfat    umask=0077      0       1
/dev/mapper/nvme0n1p1_crypt /home           ext4    defaults        0       2
/swapfile                                 none            swap    sw              0       0
/dev/disk/by-uuid/aa5fe761-4ef7-45ed-9b9f-98fab6f2a51e /data auto nosuid,nodev,nofail,x-gvfs-show 0 0

rainer@cslbox:~ $ blkid
/dev/nvme0n1p1: UUID="A923-55B5" TYPE="vfat" PARTLABEL="EFI System Partition" PARTUUID="c92e16f0-610d-4bc4-90be-96539770b81e"
/dev/nvme0n1p2: UUID="32ba72e3-26fe-441d-9190-38165e662d4c" TYPE="ext4" PARTUUID="70aebed4-ded9-4ff0-9fb0-1ebdf04790bd"
/dev/nvme1n1p1: UUID="fc6cc1a1-50d6-46af-9940-cff9839eff90" TYPE="crypto_LUKS" PARTUUID="4c27dd25-d8eb-4dea-901c-17004dda0eeb"
/dev/sda: LABEL="DATA" UUID="aa5fe761-4ef7-45ed-9b9f-98fab6f2a51e" TYPE="ext4"

```

**Comment by rsi9x** [comment]
It's interesting that it's happening with a single volume for you.  
I haven't tried this, but it has been reported on the forums that you can work around this issue by disabling the graphical boot (plymouth) by removing 'quiet splash' from the kernel boot parameters.  There seems to be something wrong with plymouth's handling of the passphrase prompt.

**Comment by rsi9x** [comment]
@feikede 
```
UUID=32ba72e3-26fe-441d-9190-38165e662d4c /               ext4    errors=remount-ro 0       1
/dev/mapper/nvme0n1p1_crypt /home           ext4    defaults        0       2

```
Based on the above, it looks like your LUKS volume is not the OS volume.   I think the problem only affects non-OS volumes.

It appears that the issue has something to do with how plymouth identifies that it should prompt for a passphrase during boot; it appears to only work if it's the OS volume.  It seems that plymouth is having a problem identifying when to prompt for password for additional volumes after the OS starts booting, so it just hangs instead.

I tried removing 'quiet splash' from the kernel parameters and it worked for me; I got a passphrase prompt for the second volume.

**Comment by feikede** [comment]
@rsi9x Thx, can confirm that it works without the splash - get the passphrase prompt for /home at console.

**Comment by clefebvre** [comment]
Hi,

We found the cause of the issue today. It's caused by the fact that your system is not merged but that the build (which uses AC_PROG_PATH) happened in a merged system.

To solve the issue:

`apt install usrmerge`

Press Y when asked whether to merge the system.

**Comment by clefebvre** [comment]
We'll send a package update also.

**Comment by rsi9x** [comment]
@clefebvre 

>It's caused by the fact that your system is not merged but that the build (which uses AC_PATH_PROG) happened in a merged system

Thanks for identifying the cause of this issue.

> We'll send a package update also.

For clarity, does this mean that you're going deploy an update that will merge all Mint systems, or are you going to fix the issue that causes it to break on unmerged systems?

**Comment by clefebvre** [comment]
We don't really know yet if we'll merge all systems. What we'll do for sure is send an update for plymouth which works on all systems and we'll build a tool for us to detect these kind of issues.

**Comment by clefebvre** [comment]
From a user point of view though, especially in light of what's happened here, I would definitely recommend people merge their systems.

**Comment by clefebvre** [comment]
Fixed in plymouth 0.9.4+mint2+ulyssa.
